roles/benchmark/tasks/run.yml

```diff
@@ -1,3 +1,11 @@
+- local_action: 
+    module: copy 
+    content: "{{ arguments | to_json }}"
+    dest: "/tmp/arguments.json"
+
+- name: Load server arguments
+  ansible.builtin.include_vars:  "/tmp/arguments.json"
```

Comment:
Hrrrmmm, let me see. The `set_fact` would need to accept a JSON variable as input.

Reply:
You don't need to. It is already in the ansible variable. You don't need the local_action above either. Just do a loop on the arguments variable which should be a dict at this point.

```yaml
  ansible.builtin.set_fact:
    "{{ item.key }}" : "{{ item.value }}"
  loop: "{{ tag_data | dict2items }}"
```

Reply:
It almost works, but the `set_fact` seems to convert everything to string, which causes issues when parsing the templates.

Reply:
I think the benchmark template can do the cast directly, no?

Reply:
Btw we need to do something to fix it because I also can't pass `-e read_ratio=0.8 -e write_ratio=0.2` to ansible-playbook without getting the same error.

Reply:
Yeah, I change benchmark to be `write_ratio | float` and the same with read_ratio in all the places I can pass a command line argument as well, which will also fix the cast issue you had. We should probably do that for any arguments that may be overridden by the user.

Reply:
That's great. I've updated the benchmark adding a cast to all the numeric parameters.